instance/ansible.py

```diff
@@ -67,59 +68,96 @@ def dict_merge(dict1, dict2):
     return dict1
 
 
-@contextmanager
-def string_to_file_path(string):
+def string_to_file_path(string, root_dir=None):
     """
-    Store a string in a temporary file, to pass on to a third-party shell command as a file parameter
-    Returns the file path string
+    Store a string in a temporary file
     """
-    fd, file_path = mkstemp(text=True)
-    fp = os.fdopen(fd, 'w')
-    fp.write(string)
-    fp.close()
-    yield file_path
-    os.remove(file_path)
+    f = NamedTemporaryFile('w', delete=False, dir=root_dir)
+    f.write(string)
+    f.close()
+    return f.name
 
 
 @contextmanager
-def run_playbook(requirements_path, inventory_str, vars_str, playbook_path, playbook_name, username='root'):
+def create_temp_dir():
     """
-    Runs ansible-playbook in a dedicated venv
+    A context manager that creates a temporary directory, returns it. Directory is deleted upon context manager exit.
+    """
+    try:
+        temp_dir = mkdtemp()
+        yield temp_dir
+    finally:
+        # This is really a "best effort" solution, if something will block the deletion (like file being locked somehow)
+        # this should just move on rather than raise.
+        shutil.rmtree(temp_dir, ignore_errors=True)
```

Comment:
This is now logged, however messages do not appear in instance logs, to make them appear in instance logs, I'd have to somehow pass `{"obj": instance.pk}` as a logging context, and instance is nowhere near this method. This can be done, for example by storing instance pk in a threadlocal varibable, or passing it though couple of levels of call stack, but I think it is not worthwile.